hal/src/stm32f2xx/spi_hal.c

```diff
@@ -644,6 +649,9 @@ void HAL_SPI_Info(HAL_SPI_Interface spi, hal_spi_info_t* info, void* reserved)
         info->mode = spiState[spi].mode;
         info->bit_order = spiState[spi].SPI_InitStructure.SPI_FirstBit == SPI_FirstBit_MSB ? MSBFIRST : LSBFIRST;
         info->data_mode = spiState[spi].SPI_InitStructure.SPI_CPOL | spiState[spi].SPI_InitStructure.SPI_CPHA;
+        if (info->version >= HAL_SPI_INFO_VERSION_2) {
+            info->ss_pin = spiState[spi].SPI_SS_Pin;
```

Comment:
Any implications on `spi_ensure_configured()` from wiznetif.cpp?

Reply:
Did you mean that the SS pin might be temporarily modified by wiznetif.cpp, so that we get the wrong pin?

Reply:
I think we are going to now always be calling `SPI.begin()` due to this change:
https://github.com/particle-iot/device-os/blob/develop/hal/network/lwip/wiznet/wiznetif.cpp#L120

Reply:
As we discussed, we should rework the wiznetif.cpp to store the SPI configuration before begin() and restore before end() to make sure the system firmware doesn't affect user's SPI configuration. But does this change here has any potential defect? This function gets the current SPI info and it does what we want.

Reply:
We are already doing that, if you take a look at `reg_wizchip_cris_cbfunc()` call. It should work fine, as I was looking at the stm32f2xx change. Resolving this.